blockchain/event_filter.go

```diff
@@ -196,3 +197,29 @@ func (e *EventFilter) appendBlockEvents(matchedEventsSofar []*FilteredEvent, hea
 	}
 	return matchedEventsSofar, processedEvents, nil
 }
+
+func (e *EventFilter) matchesEventKeys(eventKeys []*felt.Felt, keysMap []map[felt.Felt]bool) bool {
+	// short circuit if event doest have enough keys
+	for i := len(eventKeys); i < len(keysMap); i++ {
+		if len(keysMap[i]) > 0 {
+			return false
+		}
+	}
+
+	/// e.keys = [["V1", "V2"], [], ["V3"]] means:
+	/// ((event.Keys[0] == "V1" OR event.Keys[0] == "V2") AND (event.Keys[2] == "V3")).
+	//
+	// Essentially
+	// for each event.Keys[i], (len(e.keys[i]) == 0 OR event.Keys[i] is in e.keys[i]) should hold
+	for index, eventKey := range eventKeys {
+		// empty filter keys means match all
+		if index >= len(keysMap) || len(keysMap[index]) == 0 {
+			break
+		}
+		if found := keysMap[index][*eventKey]; !found {
```

Comment:
Do we check that request contains valid event keys? i.e. not nil

Reply:
`jsonrpc` package does that before our handler is called

Reply:
Hmm strange I got panic with this curl request:
```
curl --location --request POST 'http://localhost:6060' \
--header 'Content-Type: application/json' \
--data-raw '{
	"jsonrpc":"2.0",
	"method":"starknet_getEvents",
	"params": [{
        "chunk_size": 2,
        "keys": [
            [null, null]
        ]
    }],
	"id":2
}'
```